src/main/java/org/wildfly/security/auth/server/SubjectUtil.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2016 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wildfly.security.auth.server;
+
+import java.security.Principal;
+import java.security.acl.Group;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.HashSet;
+import java.util.Set;
+
+import javax.security.auth.Subject;
+
+import org.wildfly.common.Assert;
+import org.wildfly.security.auth.principal.NamePrincipal;
+import org.wildfly.security.authz.Attributes;
+import org.wildfly.security.authz.AuthorizationIdentity;
+import org.wildfly.security.authz.MapAttributes;
+import org.wildfly.security.credential.PublicKeyCredential;
+
+/**
+ * Utilities for dealing with {@link Subject}.
+ *
+ * @author <a href="mailto:sguilhen@redhat.com">Stefan Guilhen</a>
+ */
+public final class SubjectUtil {
+
+    /**
+     * Converts the supplied {@link SecurityIdentity} into a {@link Subject}.
+     *
+     * @param securityIdentity the {@link SecurityIdentity} to be converted.
+     * @return the constructed {@link Subject} instance.
+     */
+    public static Subject fromSecurityIdentity(final SecurityIdentity securityIdentity) {
+        Assert.checkNotNullParam("securityIdentity", securityIdentity);
+        Subject subject = new Subject();
+        subject.getPrincipals().add(securityIdentity.getPrincipal());
+
+        // add the 'Roles' group to the subject containing the identity's mapped roles.
+        Group rolesGroup = new SimpleGroup("Roles");
```

Comment:
I don't mind using it for compatibility with PB use cases since PB is being rolled off.  I wouldn't want to use it for JCA though since that's ongoing, and that API has been deprecated since Java 1.2.  With the new deprecation policy of Java 9, there's a good chance these APIs could be gone by 10.

We may need to look at how other JAAS implementations bundle groups to see if there's a more modern approach that we could purloin.